network/common/src/main/java/org/apache/spark/network/client/TransportClient.java

```diff
@@ -85,6 +86,25 @@ public SocketAddress getSocketAddress() {
   }
 
   /**
+   * Returns the ID used by the client to authenticate itself when authentication is enabled.
+   *
+   * @return The client ID, or null if authentication is disabled.
```

Comment:
Technically yes, but I'm pretty sure code cannot get a `TransportClient` handle before the auth bootstraps have run.